astropy/io/ascii/latex.py

```diff
@@ -384,7 +384,12 @@ def start_line(self, lines):
 
     def write(self, lines):
         lines.append(self.data_start)
+        lines_length_initial = len(lines)
         core.BaseData.write(self, lines)
+        # To remove extra space and // appended which creates an extra new line
+        # in the end.
+        if len(lines) > lines_length_initial:
```

Comment:
What about a hybrid of using the regex within the existing `if` statement.  I always get nervous about things like `[:-3]` since that is just begging to break if anything changes upstream.  I know it's not likely in this case, but in theory someone could make a sub-class and change `join` so that it adds just `r'\\'` (without the space).

Also, I guess you would want the regex to be `r'\s* \\ \s* $'` (with re.VERBOSE).